lib/jekyll/converters/markdown/kramdown_parser.rb

```diff
@@ -46,6 +45,11 @@ def convert(content)
         end
 
         private
+        def symbolize_keys(hash)
+          hash.reduce({}) { |hsh, (k, v)| hsh.merge(k.to_sym => v) }
```

Comment:
Readability wise this should be be a `do...end` block, not a `{}` block.

In theory `Hash[hash.keys.map(&:to_sym).zip(hash.values)]` will be faster, since it should be a single depth hash, and you aren't really nesting it, that said, merge is slow, if you really want it to be a block, it should be `each_with_object` which is way faster than reduce when you want to work the way you want.  Reduce is only marginally faster than `each_with_object` and only when you are using fast methods like `:<<` and others, but when you are merging (which is expensive to begin with) it's hella slower.

Reply:
This method is based on https://github.com/jekyll/jekyll/blob/621df9db8e72cfa9706ccd52b86a590bcd5d7cc3/lib/jekyll/configuration.rb#L103-L108

Can that method be optimized too, considering the depth of nesting involved..?

Reply:
@ashmaroli yessum, in a scenario (like a full-stack application) I would never reduce on a hash like that, because it's just to slow and TTFB matters a lot to me, and `reduce`+`#merge` is slow compared to `each_with_object`+`#[]=`.  It could provide *some* perf to Jekyll but probably not enough to matter unless that method is used everywhere.

Reply:
In general though, I would have just accepted the performance penalty of being indifferent with ActiveSupport though, because at least then I don't have to deal with all this logic non-sense, conversion non-sense, and all I need to worry about is shipping source.